sys/riscv/riscv/trap.c

```diff
@@ -188,12 +188,15 @@ cpu_fetch_syscall_args(struct thread *td)
 
 #include "../../kern/subr_syscall.c"
 
+/*
+ * This cannot use _CHERI_PRINTF_CAP_ARG due to the casts (the trapframe stores
+ * uintcap_t rather than void * __capability).
+ */
 #if __has_feature(capabilities)
-/* This cannot use _CHERI_PRINTF_CAP_ARG due to the cast for purecap. */
 #ifdef __CHERI_PURE_CAPABILITY__
```

Comment:
We should probably add a way to detect whether the compiler allows capabilities to be passed to varargs in hybrid mode. Maybe something like this?
```suggestion
#if defined(__CHERI_PURE_CAPABILITY__) || __has_feature(capabilities_in_hybrid_varargs)
```

Reply:
Or we just call it a flag day?

Reply:
My PR makes Clang accept `void * __capability *` too given the calling convention is the same, so whilst there will eventually be a flag day it won't be a hard flag day and instead just a "delete the workaround from CheriBSD once the newer compiler is sufficiently old".